### ISSUE TITLE
Virtual machine crashing on starting krunvm

### ISSUE THREAD

**Issue by rajasaur** [containers/krunvm#3]
Environment:
MacOS with VirtualBox 6.1.18 hosting Ubuntu 20.04 with Guest additions enabled

1. Installed libkrunfw from source
2. Installed libkrun from source.
3. git clone krunvm and build 

Normally running it with `LD_LIBRARY_PATH=/usr/local/lib64/:$LD_LIBRARY_PATH ./target/release/krunvm list ` throws a `Please re-run krunvm inside a "buildah unshare" session`

So I do a `buildah unshare` and then create the VM with
`LD_LIBRARY_PATH=/usr/local/lib64/:$LD_LIBRARY_PATH ./target/release/krunvm create debian --name deb-test`
This creates the image which can be verified with krunvm list

When trying to execute this with
`LD_LIBRARY_PATH=/usr/local/lib64/:$LD_LIBRARY_PATH ./target/release/krunvm start deb-test /bin/bash`
the Virtualbox VM completely crashes. Looking at the VBox log file just has this:

`
00:01:35.155450 emR3Debug: rc=VERR_EM_INTERNAL_ERROR
`

Not sure if Im making some basic errors, but pls let know if you need any more debug logs.

**Comment by slp** [comment]
Thanks for the info. This one may be hard to debug, given the number of moving parts (macOS->VirtualBox/NestedVirt->Ubuntu/KVM->krunvm). Also, I don't have an Intel-based Mac, so I can't test this myself.

Have you tried running a conventional KVM-based VM with some other tool (such as GNOME Boxes) inside the Ubuntu instance? Does that work?

**Comment by rajasaur** [comment]
Thanks for the quick response. Ill try it on a Linux machine to see how it goes. Ill post back results once i try there.

I built krunvm with `cargo build --release`, do you think there could be additional info if i run it without the --release flag (Sorry, just learning Rust)

**Comment by slp** [comment]
> I built krunvm with `cargo build --release`, do you think there could be additional info if i run it without the --release flag (Sorry, just learning Rust)

Looks like the immediate outcome is an abrupt crash of the VirtualBox instance, so I guess `krunvm` won't have an opportunity to produce any output, not even with a debug build.

**Comment by rajasaur** [comment]
Just a minor update. It looks like my VM did not have vhost_vsock or vhost_net modules enabled. I did try krunvm again after enabling the modules, and it still crashed. (Checked using the kata-containers check script). 

Im going to try it on a ubuntu machine next.

**Comment by rajasaur** [comment]
@slp Looks like the issue was VirtualBox. I tried with some other kvm based setups inside VirtualBox and it also crashed with a similar error.  On the new macbook pros, the "Enable nested vt-x" option was always disabled and I always had to do a `modifyvm nested-hw-virt on` to get it to show, but never figured if it actually allowed nested virtualization to work.. Searching a few virtualbox forums also indicated that there could be a virtualbox issue.

So I downloaded VMWare fusion, same setup, ubuntu 20.04.1 and same setup and it worked like a charm. There were no issues at all and I was able to run krunvm without any issues. 

Ill close this issue. Sorry for the noise earlier
